File: AlloyRunner.java
import edu.mit.csail.sdg.alloy4.*;
import edu.mit.csail.sdg.alloy4compiler.ast.*;
import edu.mit.csail.sdg.alloy4compiler.parser.*;
import edu.mit.csail.sdg.alloy4compiler.translator.*;



import java.io.*;
import java.util.List;

public class AlloyRunner {

    public static void main(String[] args) throws Err, IOException {
        // Check if the file name is provided as argument
        if (args.length != 1) {
            System.out.println("Usage: java AlloyRunner <filename>");
            return;
        }

        String fileName = args[0];
        File file = new File(fileName);

        // Check if the file exists
        if (!file.exists()) {
            System.out.println("File " + fileName + " does not exist!");
            return;
        }

        // Check if the file extension is ".als"
        if (!fileName.endsWith(".als")) {
            System.out.println("File " + fileName + " is not an Alloy file (.als)!");
            return;
        }

        // Compile the Alloy file using the Alloy Compile API
        System.out.println("Compiling " + fileName + "...");
        CompUtil.initOriginalYicesPath(null);
        CompModule module = CompUtil.parseEverything_fromFile(null, null, fileName);

        // Run all the commands and save the output to separate files
        List<Command> commands = module.getAllCommands();
        for (Command command : commands) {
            String commandName = command.toString().replaceAll("[^a-zA-Z0-9.-]", "_");
            System.out.println("Running command: " + commandName);
            A4Solution solution = TranslateAlloyToKodkod.execute_command(null, module.getAllReachableSigs(), command, null);
            String output = solution.toString();
            String outputFileName = commandName + ".txt";
            System.out.println("Saving output to file: " + outputFileName);
            writeToFile(outputFileName, output);
        }

        System.out.println("All commands completed successfully.");
    }

    private static void writeToFile(String fileName, String content) throws IOException {
        BufferedWriter writer = new BufferedWriter(new FileWriter(fileName));
        writer.write(content);
        writer.close();
    }

}

File: Alloy Specs/arepair/arepair_num_1/alloy_runner.py
import sys
import os
from edu.mit.csail.sdg.alloy4whole import SimpleGUI
from edu.mit.csail.sdg.alloy4compiler.parser import CompUtil
from edu.mit.csail.sdg.alloy4compiler.translator import TranslateAlloyToKodkod

# Check if a file path was provided as argument
if len(sys.argv) != 2:
    print("Usage: python alloy_runner.py file_path.als")
    sys.exit(1)

file_path = sys.argv[1]

# Check if the file exists
if not os.path.isfile(file_path):
    print("File not found.")
    sys.exit(1)

# Load the Alloy specification
model = CompUtil.loadModelFromFile(None, file_path)

# Translate the Alloy specification to Kodkod
options = TranslateAlloyToKodkod.getOptions()
options.solver = "MiniSatProverJNI"
kkresult = TranslateAlloyToKodkod.execute_command(None, model.getAllCommands(), options)

# Save the output in a text file
with open("output.txt", "w") as f:
    f.write(str(kkresult))

print("Output saved in output.txt")
